### PR TITLE
calculate intercept when fitting, discard outliers from dataset

### PR DESCRIPTION
**Pull request by farhadrgh**
655343d: The outlier energy conformers can be discarded from the dataset prior to training as in [ANITools](https://github.com/Jussmith01/ANI-Tools/blob/9edc5a6a3740c0a93834effe15cdc6eecbaf3961/lib/anitraintools.py#L876)

0d4a67e: The self atomic energies were calculated by linear fitting while `fit_intercept=False`. This way the average of energies used for training is not zero. Here we add the option to calculate and use the intercept to shift the molecular energies to achieve `true_energies.mean() = 0.0`. The default however is `fit_intercept=False`

### CODE REVIEW

File: torchani/utils.py [hunk 1]
```diff
@@ -181,9 +184,14 @@ def sae(self, species):
             :class:`torch.Tensor`: 1D vector in shape ``(conformations,)``
             for molecular self energies.
         """
+        intercept = 0.0
+        present_species_ = present_species(species)
+        if self.self_energies.numel() == present_species_.numel() + 1:
```
Comment (zasdfgbnm) [inline torchani/utils.py]: I think the problem is here:
Consider CH3OH, then `present_species_.numel()` will be `3`, but there should not be intercept.

Comment (farhadrgh) [reply torchani/utils.py]: Wow! good catch! I will work on it, probably making `fit_intercept` an instance variable.

File: torchani/utils.py [hunk 2]
```diff
@@ -144,16 +144,16 @@ class EnergyShifter(torch.nn.Module):
             be in order, i.e. ``self_energies[i]`` should be atom type ``i``.
```
Comment (zasdfgbnm) [inline torchani/utils.py]: Could you also add a doc for `fit_intercept` here?

Comment (farhadrgh) [reply torchani/utils.py]: Done!

**Pull request merged**